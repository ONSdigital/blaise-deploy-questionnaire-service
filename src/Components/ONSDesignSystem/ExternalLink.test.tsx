import React from "react";
import Enzyme, {render, shallow} from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import ExternalLink from "./ExternalLink";

describe("External Link Test", () => {
    Enzyme.configure({adapter: new Adapter()});

    const Props = {
        text: "Click Me",
        link: "/link",
        ariaLabel: "Aria label description"
    };

    it("matches Snapshot", () => {
        const wrapper = render(<ExternalLink {...Props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const wrapper = shallow(<ExternalLink {...Props}/>);
        expect(wrapper.exists()).toEqual(true);
    });

    it("should render with the correct text displayed", () => {
        const wrapper = shallow(<ExternalLink {...Props}/>);
        const children = wrapper.find("a");
        expect(children.html()).toContain(Props.text);
    });

    it("should render with the correct href passed in", () => {
        const wrapper = shallow(<ExternalLink {...Props}/>);
        const href = wrapper.find("a").props().href;
        expect(href).toEqual(Props.link);
    });

    it("should render with the correct aria label passed in", () => {
        const wrapper = shallow(<ExternalLink {...Props}/>);
        const ariaLabel = wrapper.find("a").props()["aria-label"];
        expect(ariaLabel).toEqual(Props.ariaLabel);
    });


});
