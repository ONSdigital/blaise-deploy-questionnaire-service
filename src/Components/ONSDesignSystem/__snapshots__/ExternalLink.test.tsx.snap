// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`External Link Test matches Snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-label": "Aria label description",
      "class": "external-link",
      "href": "/link",
      "rel": "noopener noreferrer",
      "target": "_blank",
    },
    "children": Array [
      Object {
        "data": "Click Me",
        "next": Object {
          "attribs": Object {
            "class": "svg-icon",
            "viewBox": "0 0 12 12",
            "xmlns": "http://www.w3.org/2000/svg",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "d": "M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z",
                "transform": "translate(-2 -1.99)",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": Object {
                "attribs": Object {
                  "d": "M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z",
                  "transform": "translate(-2 -1.99)",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "transform": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "transform": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "transform": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "transform": undefined,
              },
            },
            Object {
              "attribs": Object {
                "d": "M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z",
                "transform": "translate(-2 -1.99)",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "d": "M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z",
                  "transform": "translate(-2 -1.99)",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "transform": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "transform": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "transform": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "transform": undefined,
              },
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "viewBox": undefined,
            "xmlns": "http://www.w3.org/2000/xmlns/",
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "viewBox": undefined,
            "xmlns": "",
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
      Object {
        "attribs": Object {
          "class": "svg-icon",
          "viewBox": "0 0 12 12",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "d": "M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z",
              "transform": "translate(-2 -1.99)",
            },
            "children": Array [],
            "name": "path",
            "namespace": "http://www.w3.org/2000/svg",
            "next": Object {
              "attribs": Object {
                "d": "M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z",
                "transform": "translate(-2 -1.99)",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "transform": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "transform": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "d": undefined,
              "transform": undefined,
            },
            "x-attribsPrefix": Object {
              "d": undefined,
              "transform": undefined,
            },
          },
          Object {
            "attribs": Object {
              "d": "M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z",
              "transform": "translate(-2 -1.99)",
            },
            "children": Array [],
            "name": "path",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "d": "M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z",
                "transform": "translate(-2 -1.99)",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "transform": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "transform": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "d": undefined,
              "transform": undefined,
            },
            "x-attribsPrefix": Object {
              "d": undefined,
              "transform": undefined,
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "data": "Click Me",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "viewBox": undefined,
          "xmlns": "http://www.w3.org/2000/xmlns/",
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "viewBox": undefined,
          "xmlns": "",
        },
      },
    ],
    "name": "a",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-label": undefined,
      "class": undefined,
      "href": undefined,
      "rel": undefined,
      "target": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-label": undefined,
      "class": undefined,
      "href": undefined,
      "rel": undefined,
      "target": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
